community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/TransactionBoundQueryContext.scala

```diff
@@ -134,27 +133,14 @@ final class TransactionBoundQueryContext(graph: GraphDatabaseAPI,
   def exactIndexSearch(index: IndexDescriptor, value: Any) =
     JavaConversionSupport.mapToScala(statement.readOperations().nodesGetFromIndexLookup(index, value))(nodeOps.getById)
 
-  override def rangeIndexSearch(index: IndexDescriptor, value: Any) = {
-    // TODO: Push this down into kernel
-    value match {
-      case StringSeekRange(range) =>
-        range match {
-          case LowerBounded(lower) =>
-            val allNodesInIndex = JavaConversionSupport
-              .mapToScala(statement.readOperations().nodesGetFromIndexScan(index))(nodeOps.getById)
-            val readOps = statement.readOperations()
-            val propertyKeyId = index.getPropertyKeyId
-            val bound = lower.endPoint
-            allNodesInIndex.filter { (node: Node) =>
-              val nodeId = node.getId
-              readOps.nodeGetProperty(nodeId, propertyKeyId).value() match {
-                case s: String => s.startsWith(bound)
-                case _ => false
-              }
-            }
-        }
-      case _ => Iterator.empty
-    }
+  def rangeIndexSearch(index: IndexDescriptor, value: Any) = value match {
```

Comment:
@davidegrohmann This is the way it looks now. (Your line comment disappeared because it was made on the commit rather than on the PR.)